test/bricks-net-api.test.js

```diff
@@ -0,0 +1,634 @@
+		assert.equal('function', typeof api.GET);
+	});
+	
+	it('should export `POST` function', function () {
+		assert.equal('function', typeof api.POST);
+	});
+	
+	it('should export `POSTFromFile` function', function () {
+		assert.equal('function', typeof api.POSTFromFile);
+	});
+	
+	it('should export `HTTP` function', function () {
+		assert.equal('function', typeof api.HTTP);
+	});
+	
+	it('should export `HTTPResponse` function', function () {
+		assert.equal('function', typeof api.HTTPResponse);
+	});
+	
+	describe('`GET` function', function () {
+		it('should return an instance of `GET` class', function () {
+			assert.equal(true, api.GET("url") instanceof api.GET);
+		});
+		
+		it('should set `url`', function () {
+			assert.equal('url', api.GET("url").url);
+		});
+		
+		it('should `AllowRedirects` return an instance of `GET` class', function () {
+			assert.equal(true, api.GET("url").AllowRedirects() instanceof api.GET);
+		});
+		
+		it('should `AllowRedirects` set `allow_redirects`', function () {
+			assert.equal(true, api.GET("url").AllowRedirects().allow_redirects);
+			assert.equal(false, api.GET("url").AllowRedirects(false).allow_redirects);
+			assert.equal(true, api.GET("url").AllowRedirects(true).allow_redirects);
+		});
+		
+		it('should `UserAgent` return an instance of `GET` class', function () {
+			assert.equal(true, api.GET("url").UserAgent("USERAGENT") instanceof api.GET);
+		});
+		
+		it('should `UserAgent` set `custom_user_agent`', function () {
+			assert.equal("USERAGENT", api.GET("url").UserAgent("USERAGENT").custom_user_agent);
+			assert.equal("", api.GET("url").UserAgent("").custom_user_agent);
+		});
+		
+		it('should `UserAgent` throw if argument not set', function () {
+			assert.throws(function () {
+				api.GET("url").UserAgent();
+			});
+		});
+	});
+	
+	describe('`POST` function', function () {
+		it('should return an instance of `POST` class', function () {
+			assert.equal(true, api.POST("url") instanceof api.POST);
+		});
+		
+		it('should throw on invalid arguments', function () {
+			assert.throws(function () {
+				api.POST();
+			});
+			assert.throws(function () {
+				api.POST(1, 3, 5);
+			});
+			assert.throws(function () {
+				api.POST("url", "body", "content/type", "invalid argument");
+			});
+		});
+	});
+	
+	describe('`POSTFromFile` function', function () {
+		it('should return an instance of `POSTFromFile` class', function () {
+			assert.equal(true, api.POSTFromFile("url", __filename, "application/javascript") instanceof api.POSTFromFile);
+		});
+	});
+	
+	describe('`HTTP` function', function () {
+		var KBYTES_IN_BYTES = 1000;
+		var MBYTES_IN_BYTES = 1000 * KBYTES_IN_BYTES;
+		var largeBodyLength = 50 * MBYTES_IN_BYTES;
+		var largeBody;
+		
+		var http = require('http');
+		var server;
+		var sockets = [];
+		
+		var serverRequestHandler;
+		
+		var _DEBUG_LOG = function () {};
+		//var _DEBUG_LOG = function () { _DEBUG_LOG.apply(console, arguments); };
```

Comment:
Yes, these tests did require some debugging. Wrt the global flag - I had to disable/enable the logging more granularly (by commenting and uncommenting).